src/Raven.Server/Documents/Sharding/Subscriptions/OrchestratedSubscriptionConnection.cs

```diff
@@ -73,22 +82,23 @@ protected override void GatherIncludesForDocument(OrchestratorIncludesCommandImp
             // no op
         }
 
-        protected override async Task<bool> WaitForChangedDocsAsync(AbstractSubscriptionConnectionsState state, Task pendingReply)
+        protected override async Task<bool> WaitForChangedDocsAsync(AbstractSubscriptionConnectionsState state)
         {
             // nothing was sent to the client, but we need to let the shard know he can continue
-            await NotifyShardAboutBatchCompletion();
+            await NotifyShardAboutBatchCompletionAsync(orchestratorCv: null);
 
-            return await base.WaitForChangedDocsAsync(state, pendingReply);
+            return await base.WaitForChangedDocsAsync(state);
         }
 
-        private async Task NotifyShardAboutBatchCompletion()
+        private async Task NotifyShardAboutBatchCompletionAsync(string orchestratorCv)
         {
             var batch = _processor.CurrentBatch;
 
             if (batch != null)
             {
                 // let sharded subscription worker know that we sent the batch to the client and received an ack request from it
                 batch.SendBatchToClientTcs.TrySetResult();
+                batch.LastSentChangeVectorInBatch = orchestratorCv; //pass the orchestrator merged change vector to shard, so it will ack it
```

Comment:
it seems like a precaution for empty batch received on orchestrator (so we will advance the shard),
according to the code if its null then in `AcknowledgeSubscriptionBatchCommand` the `currentState.ShardingState.ChangeVectorForNextBatchStartingPointForOrchestrator` will stay the same which is fine (since this was empty batch)